clients/spark/core/src/main/scala/io/treeverse/clients/GarbageCollector.scala

```diff
@@ -301,20 +302,33 @@ object GarbageCollector {
   }
 
   def main(args: Array[String]) {
-    if (args.length != 2) {
+    if (args.length < 1) {
       Console.err.println(
-        "Usage: ... <repo_name> <region>"
+        "Usage: ... <repo_name> [region]"
       )
       System.exit(1)
     }
 
     val spark = SparkSession.builder().appName("GarbageCollector").getOrCreate()
+    val hc = spark.sparkContext.hadoopConfiguration
+
+    val apiURL = hc.get(LAKEFS_CONF_API_URL_KEY)
+    val accessKey = hc.get(LAKEFS_CONF_API_ACCESS_KEY_KEY)
+    val secretKey = hc.get(LAKEFS_CONF_API_SECRET_KEY_KEY)
+    val apiClient = new ApiClient(apiURL, accessKey, secretKey)
+    val storageType = apiClient.getBlockstoreType()
+
+    if (storageType == StorageTypeS3 && args.length != 2) {
+      Console.err.println(
+        "Usage: ... <repo_name> <region>"
+      )
+      System.exit(1)
+    }
```

Comment:
This does not verify `args.length == 1` when `storageType != StorageTypeS3`.  There are several ways forward: we could [slice](https://www.scala-lang.org/api/2.13.3/scala/Array.html#slice(from:Int,until:Int):Array[A]) off the first element each time we use it (into a new Array) and then verify that nothing is left.

Or we could change the CLI and use a real flags library.  I would prefer this option.  And it will take us less time if we keep the current PR and just open an issue to clean things up.